server/src/main/java/org/eclipse/lsp/cobol/core/MessageServiceParser.java

```diff
@@ -152,6 +152,31 @@ protected void validateStartsWith(String input, String... startsWith) {
     }
   }
 
+  /**
+   * Validate string length against range and throw an error if it is incorrect
+   *
+   * @param input string to check
+   * @param minLength allowed minimum length
+   * @param maxLength allowed maximum length
+   */
+  protected void validateStringLengthRange(String input, Integer minLength, Integer maxLength) {
+    if (input != null && !(input.length() >= minLength && input.length() <= maxLength)) {
+      notifyError("parsers.stringLengthRange", minLength.toString(), maxLength.toString());
+    }
+  }
+
+  /**
+   * Validate allowed string values and throw an error if it is incorrect
+   *
+   * @param input string to check
+   * @param allowedValues arrays of allowed starting string values for Input parameter
+   */
+  protected void validateAllowedValues(String input, String... allowedValues) {
+    if (input != null && !checkInputInAllowedValues(input, allowedValues)) {
+      notifyError("parsers.allowedStringValues", String.join(",", allowedValues));
```

Comment:
Maybe, it would be better to join the values with `", "` for better readability?

Reply:
Done.